Formula/hdf5.rb

```diff
@@ -58,7 +58,7 @@ def install
     system "./configure", *args
 
     # Avoid shims in settings file
-    inreplace "src/libhdf5.settings", HOMEBREW_LIBRARY/"Homebrew/shims/mac/super/clang", "/usr/bin/clang"
+    inreplace "src/libhdf5.settings", %r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/#{ENV.cc}}, ENV.cc
```

Comment:
Probably need to `inreplace` both C and C++ compilers
Previous macOS variant did that (`clang` and `clang++`)